test/byContract/IexecPoco/IexecPoco1.test.ts

```diff
@@ -204,10 +223,296 @@ describe('IexecPoco1', () => {
             expect(deal.sponsor).to.equal(requester.address);
         });
 
-        it('[TODO] Should match orders with all assets and callback (Standard)', () => {});
-        it('[TODO] Should match orders without dataset (Standard)', () => {});
-        it('[TODO] Should match orders without dataset and with callback (Standard)', () => {});
+        it('[Standard] Should match orders without callback', async () => {
+            orders.requester.callback = AddressZero;
+            await depositForRequesterAndSchedulerWithDefaultPrices();
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            const dealId = getDealId(iexecWrapper.getDomain(), orders.requester, taskIndex);
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.emit(
+                iexecPoco,
+                'OrdersMatched',
+            );
+            // Check deal
+            const deal = await iexecPoco.viewDeal(dealId);
+            expect(deal.callback).to.equal(AddressZero);
+        });
+
+        it('[Standard] Should match orders without dataset', async () => {
+            orders.dataset.dataset = AddressZero;
+            orders.requester.dataset = AddressZero;
+            // Compute prices, stakes, rewards, ...
+            const dealPrice = (appPrice + workerpoolPrice) * volume; // no dataset price
+            const schedulerStake = await iexecWrapper.computeSchedulerDealStake(
+                workerpoolPrice,
+                volume,
+            );
+            // Deposit required amounts.
+            await iexecWrapper.depositInIexecAccount(requester, dealPrice);
+            await iexecWrapper.depositInIexecAccount(scheduler, schedulerStake);
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            const dealId = getDealId(iexecWrapper.getDomain(), orders.requester, taskIndex);
+            const tx = iexecPocoAsRequester.matchOrders(...orders.toArray());
+            // Check balances and frozen.
+            // Dataset price shouldn't be included.
+            await expect(tx).to.changeTokenBalances(
+                iexecPoco,
+                [iexecPoco, requester, scheduler],
+                [dealPrice + schedulerStake, -dealPrice, -schedulerStake],
+            );
+            expect(await iexecPoco.frozenOf(requester.address)).to.equal(dealPrice);
+            // Check events.
+            await expect(tx).to.emit(iexecPoco, 'OrdersMatched');
+            // Check deal
+            const deal = await iexecPoco.viewDeal(dealId);
+            expect(deal.dataset.pointer).to.equal(AddressZero);
+            expect(deal.dataset.owner).to.equal(AddressZero);
+            expect(deal.dataset.price.toNumber()).to.equal(0);
+            // Should not be impacted by datasetVolume == 0
+            expect(deal.botSize.toNumber()).to.equal(volume);
```

Comment:
Not sure to see why we need that (?)

Reply:
It's to make sure that this computation is always correct when we send a signed order with `datasetOrder.dataset == 0x0` and `datasetOrder.volume > 0`.
https://github.com/iExecBlockchainComputing/PoCo/blob/develop/contracts/modules/delegates/IexecPoco1Delegate.sol#L300

Reply:
Not sure to understand since the `_computeDealVolume` takes min values, can you elaborate?

Reply:
Do you confirm dataset order volume 321 here?